### PR TITLE
curl: progress bar refresh

### PR DESCRIPTION
**Pull request by bagder**
... for when the total transfer size is unknown. 

Also attempts to get the terminal width properly first using ioctl() now.

The `-=O=-` "ship" moves when data is transferred. The four flying "hashes" move on each refresh, independent of data.

Demo video: https://youtu.be/8QnL9-5Nl9g

### CODE REVIEW

File: src/tool_cb_prg.c [hunk 1]
```diff
@@ -31,6 +35,56 @@
...
+  buf[pos] = '#';
+  pos = sinus[(bar->tick + 5)%200] / (10000 / check);
+  buf[pos] = '#';
+  pos = sinus[(bar->tick + 10)%200] / (10000 / check);
+  buf[pos] = '#';
+  pos = sinus[(bar->tick + 15)%200] / (10000 / check);
+  buf[pos] = '#';
+
+  fputs(buf, stderr);
+  bar->tick += 2;
```
Comment (jay) [inline src/tool_cb_prg.c]: where is the check on tick it seems like this could result in signed overflow

Comment (bagder) [reply src/tool_cb_prg.c]: Indeed, I'll add a:
~~~
  if(bar->tick >= 200)
    bar->tick -= 200;

**Comment by bagder** [comment]
The programmatic way of getting the terminal width of course makes it annoying to test (since it'll vary depending on width), since I put the environment variable use as a backup instead of the other way around... Test 1148 for example.

Should I instead allow the presence of a `COLUMNS` variable use that? Then we can set that in the test suite...

**Comment by bagder** [comment]
I decided to reverse the order. Now the `COLUMNS` environment variable is checked and used if set, and only if not set curl will use `ioctl()` to figure out the *actual* width. This allows us to set `COLUMNS` to a fixed value in the test suite and make that work independently of the screen width where the tests are run!

**Comment by bagder** [comment]
Again the macos builds never start so I've killed them. The other builds seem pleased now.

**Comment by jay** [comment]
> Again the macos builds never start so I've killed them. The other builds seem pleased now.

Supposedly it was resolved (again) at some point on the 19th. https://www.traviscistatus.com/incidents/mc9x2wmpnvhg

**Comment by mercutiodesign** [comment]
It looks like the manpage is wrong now:

       -#, --progress-bar
              Make  curl  display  transfer  progress as a simple progress bar
              instead of the standard, more informational, meter.

              This progress bar draws a single line of '#'  characters  across
              the screen and shows a percentage if the transfer size is known.
              For transfers without a known size, it will instead  output  one
              '#' character for every 1024 bytes transferred.

Not sure if that is just me, but I would personally prefer the old output style over this one but it does not look like it's configurable.

**Pull request closed without merging**